### ISSUE TITLE
`ResourceLoader.load_threaded_request()` hangs for `use_sub_threads = true` with no error messages

### ISSUE THREAD

**Issue by danny88881** [godotengine/godot#85255]
### Godot version

v4.2.rc1.official [ad72de508]

### System information

Godot v4.2.rc1 - Windows 10.0.22621 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3080 (NVIDIA; 31.0.15.3758) - 11th Gen Intel(R) Core(TM) i5-11400 @ 2.60GHz (12 Threads)

### Issue description

ResourceLoader.load_threaded_request() hangs indefinitely for use_sub_threads = true with no error messages.
It happens when you have an instance of a resource in a scene and are trying to preload the resource in the script of the scene.

I think it could be deadlock and the threads are stuck busy waiting, but I'm just speculating as there is no error message and I haven't looked into the source code.

Happens in v4.1.2.stable.official [399c9dc39] and v4.2.rc1.official [ad72de508]

might be similar to https://github.com/godotengine/godot/issues/71726

### Steps to reproduce

1. Create a scene "S1" with script 
2. Create a scene "S2" with script and preload "S1" within the script
3. Create an instance of "S1" within "S2"
4. Try to call "ResourceLoader.load_threaded_request()" on "S2" with "use_sub_threads = true"

### Minimal reproduction project
[ResourceLoaderTest.zip](https://github.com/godotengine/godot/files/13445888/ResourceLoaderTest.zip)

**Comment by danny88881** [comment]
Not sure how common this issue is but I'm working around it by doing separate calls to "ResourceLoader.load_threaded_request()" for those specific instanced and preloaded scenes before loading the larger scene.

**Comment by akien-mga** [comment]
Could you test the build linked here?
https://github.com/godotengine/godot/pull/85039#issuecomment-1821055056

**Comment by danny88881** [comment]
Yeah, still stuck at 0% with no error messages.

**Comment by akien-mga** [comment]
Confirmed on Linux with latest `master` branch (ce3bac5f55507a81fb76eb0fc0b37693f663dad0).

When closing the MRP, it deadlocks. GDB gives this stacktrace:
```
(gdb) bt
#0  0x00007f8daa1b4785 in clock_nanosleep@GLIBC_2.2.5 () from /lib64/libc.so.6
#1  0x00007f8daa1b8fe3 in nanosleep () from /lib64/libc.so.6
#2  0x0000000006c05a05 in OS_Unix::delay_usec (this=0x7ffd528bfee0, p_usec=1000) at ./drivers/unix/os_unix.cpp:283
#3  0x0000000009a323ad in ResourceLoader::clear_thread_load_tasks () at ./core/io/resource_loader.cpp:1054
#4  0x000000000579a437 in Main::cleanup (p_force=false) at main/main.cpp:3765
#5  0x000000000570bc0a in main (argc=10, argv=0x7ffd528c04c8) at platform/linuxbsd/godot_linuxbsd.cpp:76
```

I confirm the issue is reproducible in 4.2-rc1, 4.1-stable and 4.0-stable, so it's not a recent regression.

**Comment by jsjtxietian** [comment]
Crawling some stucked stack trace, looks like they all wait for semaphore, it does look suspicious:

![image](https://github.com/godotengine/godot/assets/8315986/04b83a5c-6001-496f-8af6-d3ab2346883d)

**Comment by Hilderin** [comment]
Yes, I think I found the problem, but I'm not sure how to fix it.
The problem is caused because the ResourceLoader takes control of the MessageQueue but if an error occurs during the loading of resources, the message queue is beeing flushed and in this queue we got the error message from the loader, but after that, at each flush we receive the error telling us that we are not on the main thread on execute add_child from the EditorToaster.

I'm not sur if a tweak in the EditorToaster is the right move and removing the MessageQueue from the ResourceLoader is probably a bad idea.

This screenshot shows the problem clearly:
![image](https://github.com/godotengine/godot/assets/81109165/661a5437-4eba-48a3-b9e9-f5fe907627ed)
